sdk/compiler/damlc/daml-stdlib-src/DA/Crypto/Text.daml

```diff
@@ -91,6 +94,14 @@ isBytesHex hex = Text.length hex % 2 == 0 && isHex hex
 isDEREncoded : Text -> Bool
 isDEREncoded msg = isBytesHex msg && Text.substring 0 2 msg == "30" && Text.length msg == 2 * (2 + (hexByteToInt (Text.substring 2 2 msg)))
 
+#ifdef DAML_VALIDATE_CRYPTO_KEY
+-- | Validates that the public key is on the SECP256K1 curve. Public key argument should be a DER formatted hex string.
+validatePublicKey : PublicKeyHex -> Bool
+validatePublicKey pk
+  | not (isDEREncoded pk) = error $ "Public key argument is not a DER encoded hex string: " <> pk
+  | otherwise = primitive @"BESecp256k1ValidateKey" (Text.asciiToLower pk)
```

Comment:
I think we should do the validation and lower case translation in scala directly, to avoid excessive in cost in gaz.